pkg/cmd/pulumi/whoami.go

```diff
@@ -58,20 +59,50 @@ func newWhoAmICmd() *cobra.Command {
 			}
 
 			if verbose {
+				if jsonOut {
+					return outputWhoAmIJSON(WhoAmIJSON{
+						User:          name,
+						Organizations: orgs,
+						BackendURL:    b.URL(),
+					})
+				}
 				fmt.Printf("User: %s\n", name)
 				fmt.Printf("Organizations: %s\n", strings.Join(orgs, ", "))
 				fmt.Printf("Backend URL: %s\n", b.URL())
 			} else {
+				if jsonOut {
+					return outputWhoAmIJSON(WhoAmIJSON{
+						User: name,
+					})
+				}
 				fmt.Println(name)
 			}
 
 			return nil
 		}),
 	}
 
+	cmd.PersistentFlags().BoolVarP(
+		&jsonOut, "json", "j", false, "Emit output as JSON")
+
 	cmd.PersistentFlags().BoolVarP(
 		&verbose, "verbose", "v", false,
 		"Print detailed whoami information")
 
 	return cmd
 }
+
+// WhoAmIJSON is the shape of the --json output of this command.
+type WhoAmIJSON struct {
+	User          string   `json:"user"`
+	Organizations []string `json:"organizations,omitempty"`
+	BackendURL    string   `json:"backendUrl"`
+}
+
```

Comment:
@mrod-io Minor follow up as you're making the fixups:
@AaronFriel pointed out that we already have a struct we could use here instead of defining a new one:
about.go defines a "backendAbout" struct with roughly the same information.

https://github.com/pulumi/pulumi/blob/400c77127346eee8f33e3623781e22e2c270d079/pkg/cmd/pulumi/about.go#L301-L306

If we re-use that struct here, the shape will match the output of `pulumi about --json`, which is desirable.

Reply:
I went with your more recent [suggestion](https://github.com/pulumi/pulumi/pull/12374#discussion_r1128718872). You're right that using `backendAbout` means `pulumi whoami -v` would always emit a never-used field "Name", like:

```bash
$ pulumi whoami -j
{
  "name": "",
  "user": "username",
  "organizations": [
    "org_one",
    "username_org"
  ],
  "url": "https://app.pulumi.com/username"
}
```

Happy to do either, though 😃